gateway/modules/crud/batcher.go

```diff
@@ -0,0 +1,125 @@
+package crud
+
+import (
+	"context"
+	"time"
+
+	"github.com/spaceuptech/space-cloud/gateway/model"
+	"github.com/spaceuptech/space-cloud/gateway/utils"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spaceuptech/space-cloud/gateway/config"
+)
+
+// data structure for storing a group of channels
+// against particular table/collection of particular database
+type (
+	batchMap map[string]map[string]map[string]batchChannels // keys are project name, dbAlias, table name
+
+	batchChannels struct {
+		request batchRequestChan
+		close   chan struct{}
+	}
+
+	batchRequestChan chan batchRequest
+
+	batchRequest struct {
+		document interface{}
+		response chan error
+	}
+)
+
+// CloseBatchOperation closes all go routines associated with individual collection for batch operation
+func (m *Module) CloseBatchOperation() {
+	for _, projectValue := range m.batchMapTableToChan {
+		for _, dbAliasValue := range projectValue {
+			for _, tableValue := range dbAliasValue {
+				tableValue.close <- struct{}{}
+			}
+		}
+	}
+}
+
+// initBatchOperation creates go routines for executing batch operation associated with individual collection
+func (m *Module) initBatchOperation(project string, crud config.Crud) {
+	batch := batchMap{}
+	for dbAlias, dbInfo := range crud {
+		if dbInfo.Enabled {
+			for tableName, tableInfo := range dbInfo.Collections {
+				done := make(chan struct{})                                          // channel for closing go routine
+				addInsertToBatchCh := make(batchRequestChan, tableInfo.BatchRecords) // channel for adding request to batch op
```

Comment:
```suggestion
				addInsertToBatchCh := make(batchRequestChan, 20) // channel for adding request to batch op
```

Reply:
but the channel length should be equal to the length of records to be batched

Reply:
No. It has nothing to do with the batch size